Refactor twilio directions

jonleung opened the pull request.

zachlatta commented on workshops/lib/twilio-basic/signup.md:
```diff
@@ -4,185 +4,82 @@ By the end of this tutorial, you will have a Twilio account SID and
+4. Twilio then asks you to enter a phone number. Do that.
+5. Click "Text Me" and continue following remaining directions until you see
+   this screen:
+   > ![](img/finish_verification_screen.png)
```
This should have a newline before it to separate it from the above bullet point.

jonleung replied:
Why is this? If this is not in violation to the styleguide, I would prefer not to change it.

jonleung replied:
Actually I think I understand.

zachlatta commented on workshops/lib/twilio-basic/signup.md:
```diff
@@ -4,185 +4,82 @@ By the end of this tutorial, you will have a Twilio account SID and
 
-After obtaining the promo code:
+6. Then scroll all the way down:
+   > ![](img/scroll_down.gif)
```
Same thing with the newline here.

jonleung replied:
:+1:

zachlatta commented on workshops/lib/twilio-basic/signup.md:
```diff
@@ -4,185 +4,82 @@ By the end of this tutorial, you will have a Twilio account SID and
-- Enter the promo code
-- Uncheck "Turn on automatic recharge"
+7. Click the "Upgrade your account" button
+   > ![](img/upgrade_account.png)
```
And here.

jonleung replied:
:+1:

zachlatta commented on workshops/lib/twilio-basic/signup.md:
```diff
@@ -4,185 +4,82 @@ By the end of this tutorial, you will have a Twilio account SID and
 
-> ![](img/enter_promo_code.gif)
+8. You will see this screen:
+   > ![](img/upgrade_your_account.png)
```
And here.

jonleung replied:
:+1:

zachlatta commented on workshops/lib/twilio-basic/signup.md:
```diff
@@ -4,185 +4,82 @@ By the end of this tutorial, you will have a Twilio account SID and
+    - Leave the "Company/Project URL" and "Phone Number" fields blank
+    - For the credit card information, although Twilio is giving each member
+      $100 worth of credit, you still need a credit card to signup for Twilio.
+      For your convenience, we have provided a credit card to signup with below.
```
These bullet points should be indented two spaces, not four.

jonleung replied:
:+1:

zachlatta commented on workshops/lib/twilio-basic/signup.md:
```diff
@@ -4,185 +4,82 @@ By the end of this tutorial, you will have a Twilio account SID and
 
-> ![](img/click_upgrade_account.png)
+      | Field              | Value                               |
+      |--------------------|-------------------------------------|
```
The format we use for other tables is below:

```
| Field              | Value                               |
| ------------------ | ----------------------------------- |
```

jonleung replied:
I have never done this and I have lots of tables in the repository...

jonleung replied:
Therefore I don't think it makes sense to all of a sudden enforce this now especially when it's not in the style guide.

zachlatta replied:
Every table created by someone that's not you follows this format (as far
as I'm aware). I'll add this to the Markdown style guide.

On Tue, Feb 2, 2016 at 12:07 PM, Jonathan Leung <notifications@github.com>
wrote:

> In workshops/lib/twilio-basic/signup.md
> <https://github.com/hackclub/hackclub/pull/667#discussion_r51626676>:
>
> >
> > -> ![](img/click_upgrade_account.png)
> > +      | Field              | Value                               |
> > +      |--------------------|-------------------------------------|
>
> Therefore I don't think it makes sense to all of a sudden enforce this now
> especially when it's not in the style guide.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hackclub/hackclub/pull/667/files#r51626676>.
>



-- 
Zach Latta
https://zachlatta.com <http://zachlatta.com>

jonleung replied:
For now, can we merge this because I would like to do other work other than fixing markdown table formatting...

zachlatta replied:
Let's fix the formatting first because if we don't fix it now, it'll never
be fixed. And it shouldn't take long.

On Tue, Feb 2, 2016 at 1:00 PM, Jonathan Leung <notifications@github.com>
wrote:

> In workshops/lib/twilio-basic/signup.md
> <https://github.com/hackclub/hackclub/pull/667#discussion_r51634099>:
>
> >
> > -> ![](img/click_upgrade_account.png)
> > +      | Field              | Value                               |
> > +      |--------------------|-------------------------------------|
>
> For now, can we merge this because I would like to do other work other
> than fixing markdown table formatting...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hackclub/hackclub/pull/667/files#r51634099>.
>



-- 
Zach Latta
https://zachlatta.com <http://zachlatta.com>

zachlatta commented on workshops/lib/twilio-basic/signup.md:
```diff
@@ -4,185 +4,82 @@ By the end of this tutorial, you will have a Twilio account SID and
+    >  - Club members: Ask your club leader for the Twilio promo code.
+    >  - Club leaders: send an email to the email address
+        `jonathanleung1337@gmail.com` with your full name and school and ask for the
+        Twilio promo code. They will respond to you quickly.
```
Lines 52 and 53 should have `>` at the beginning of them to indicate the blockquote.

jonleung replied:
:+1:

zachlatta commented on workshops/lib/twilio-basic/signup.md:
```diff
@@ -4,185 +4,82 @@ By the end of this tutorial, you will have a Twilio account SID and
+17. Write down your account SID and authentication token somewhere that you can
+easily retrieve them. Make sure you remember which one is which. In my case,
+    - my account SID is `AC6f0013735dcb2fa4db7e35baa5c7e344`
+    - my authentication token is `fb83ceb820510cf695cf1f45b157c12f`
```
Should be indented by two spaces.

zachlatta replied:
Also, is this SID and token valid? If so, we should change them to fake values.

jonleung replied:
They are valid. However, it is a junk / test account so I think it's ok to leave them there.

zachlatta replied:
Err, let's change them to something invalid to discourage people from
taking advantage of real credentials...

On Tue, Feb 2, 2016 at 12:09 PM, Jonathan Leung <notifications@github.com>
wrote:

> In workshops/lib/twilio-basic/signup.md
> <https://github.com/hackclub/hackclub/pull/667#discussion_r51626913>:
>
> >
> > -# Feedback
> > +17. Write down your account SID and authentication token somewhere that you can
> > +easily retrieve them. Make sure you remember which one is which. In my case,
> > +    - my account SID is `AC6f0013735dcb2fa4db7e35baa5c7e344`
> > +    - my authentication token is `fb83ceb820510cf695cf1f45b157c12f`
>
> They are valid. However, it is a junk / test account so I think it's ok to
> leave them there.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hackclub/hackclub/pull/667/files#r51626913>.
>



-- 
Zach Latta
https://zachlatta.com <http://zachlatta.com>

jonleung replied:
I really don't want to retake the screenshots... I would strongly just prefer to merge this so I can focus on other work.

zachlatta replied:
Okay, in the future let's not have real credentials online...

On Tue, Feb 2, 2016 at 12:59 PM, Jonathan Leung <notifications@github.com>
wrote:

> In workshops/lib/twilio-basic/signup.md
> <https://github.com/hackclub/hackclub/pull/667#discussion_r51634017>:
>
> >
> > -# Feedback
> > +17. Write down your account SID and authentication token somewhere that you can
> > +easily retrieve them. Make sure you remember which one is which. In my case,
> > +    - my account SID is `AC6f0013735dcb2fa4db7e35baa5c7e344`
> > +    - my authentication token is `fb83ceb820510cf695cf1f45b157c12f`
>
> I really don't want to retake the screenshots... I would strongly just
> prefer to merge this so I can focus on other work.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hackclub/hackclub/pull/667/files#r51634017>.
>



-- 
Zach Latta
https://zachlatta.com <http://zachlatta.com>

zachlatta commented:
Left a bunch of comments on styling!

ad510 commented:
@zachlatta I hope you didn't learn the wrong lesson from all my nitpicks in #652. I looked at the rendered markdown and I think the only suggestion that would actually change what club members see is the one about the SID and token.

In general I try to focus on club members getting the maximum value out of workshops, and I only like to pick on little things when I think they have the potential to confuse people doing the workshops. I did talk to Jonathan over hangouts about his bigger picture approach in the personal website workshop and the workshop arc, and it seemed pretty clear to me that a top priority right now is actually getting workshops shipped, which is why I avoided making comments that would involve rewriting a bunch of the workshop. (Also he had some experience on what did and didn't work in the old portfolio workshop, so I deferred to him on that.) In the longer term I think it's at least as important to get the high level structure and organization right as it is to fix typos, and I really don't care at all about standardizing formatting where club members won't notice the difference.

That's just me though. :smile: Do whatever you think is right, and I hope Jonathan appreciates the feedback since you're the first person to comment on this PR.

ad510 commented:
Shortened my comment at https://github.com/hackclub/hackclub/pull/667#issuecomment-174739231 a little.

zachlatta commented:
@ad510: That's totally justified criticism (my comments weren't influenced
by your previous review style). Most of these are to get the document to
abide to our Markdown style guide (
https://github.com/hackclub/meta/blob/master/styleguides/markdown.md),
which has to happen before it's merged.

On Mon, Jan 25, 2016 at 7:19 PM, Andrew Downing <notifications@github.com>
wrote:

> Shortened my comment at #667 (comment)
> <https://github.com/hackclub/hackclub/pull/667#issuecomment-174739231> a
> little.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hackclub/hackclub/pull/667#issuecomment-174741778>.
>



-- 
Zach Latta
https://zachlatta.com <http://zachlatta.com>

ad510 commented:
Fair enough. I sometimes get bothered about people making a big deal about style guides because it's often an excuse for not fixing more important issues. (What kind of message would it send to club leaders if they submit a workshop they made on their own time, and they get more comments on conforming to the style guide than about their actual content?) But I suppose that's pretty far down on the list of things Hack Club should be worrying about.

jonleung commented:
@ad510, I agree for other contributors who are not staff, we shouldn't nit pick on style errors. Instead the staff should fix them. However for staff contributions, they should conform to the styleguide.

jonleung commented:
I have fixed the styling errors. I have also added my mistakes to my own little cheat sheet / checklist to try to prevent this from happening in the future!

ad510 commented:
I discussed the style guide with Zach a few days ago on Slack, so I understand the situation better now.

jonleung commented:
@ad510, would you be interested in actually taking the task of merging this pull request?

ad510 commented:
I don't have enough repo permissions. Also, how would having me merge the PR save you guys time?

paked commented:
@ad510 I think @jonleung means that you will be responsible for making sure everything is up to standard (ie. not broken) and fulfilling the initial intentions of the PR before the merge happens.

ad510 commented:
I'll take a look at it after class.

jonleung commented:
> Re: I don't have enough repo permissions. Also, how would having me merge the PR save you guys time?

@ad510 we just gave you repo permissions! And yes, this would indeed save us time because it's distributing work to more people! 

> re: @ad510 I think @jonleung means that you will be responsible for making sure everything is up to standard (ie. not broken) and fulfilling the initial intentions of the PR before the merge happens.

@ad510, if this is something you're interested in, it would actually be great if you could merge it in as well!

ad510 commented:
I opened #685 to fix a couple typos (which I would've pointed out earlier if I'd noticed them).

I still don't understand why this is assigned to me though. Can't you merge PRs with one click?
